### ISSUE TITLE
Add Jetty 11 Embedded Module

### ISSUE THREAD

**Issue by manovotn** [arquillian/arquillian-container-jetty#43]
Jetty 11 is running with Jakarta EE 9 which means we need a new module to support that.

The usual set of changes is needed - e.g. swap all `javax` deps for their `jakarta` counterparts, curate common module and so on.
On top of that, there seem to be some changes inside Jetty that will require more attention to make it work equivalently.
Namely, the following two classes vanished from source in newer version and I don't yet know replacements for them:
* [`JavaUtilLog](https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-util/src/main/java/org/eclipse/jetty/util/log/JavaUtilLog.java)
* [Configuration.ClassList](https://github.com/eclipse/jetty.project/blob/jetty-9.4.x/jetty-webapp/src/main/java/org/eclipse/jetty/webapp/Configuration.java#L99)

Both are used by `JettyEmbeddedContainer` in this project.

This change will require new Arq. core with updated servlet protocol, see https://github.com/arquillian/arquillian-core/issues/248

**Comment by manovotn** [comment]
@gregw sorry to drag you into another random issue, but could you perhaps point me to some docs/code which shows how to replace the above two classes in newer jetty versions? Eyeing Jetty 11 here because of EE 9 support (and subsequently testing on Weld).

The usage of the above classes within jetty 9 module can be seen [here](https://github.com/arquillian/arquillian-container-jetty/blob/master/jetty-embedded-9/src/main/java/org/jboss/arquillian/container/jetty/embedded_9/JettyEmbeddedContainer.java#L93) and [here](https://github.com/arquillian/arquillian-container-jetty/blob/master/jetty-embedded-9/src/main/java/org/jboss/arquillian/container/jetty/embedded_9/JettyEmbeddedContainer.java#L144).
Basically, one seems to be logging redirection, the other server class configuration. Though I am only just trying to get a hang of how this container works, so I might be wrong :)

**Comment by joakime** [comment]
Jetty `11.0.0` has been officially released.

https://github.com/eclipse/jetty.project/releases/tag/jetty-11.0.0

There is no longer a `org.eclipse.jetty.util.log.JavaUtilLog` class.
Jetty 10+ has migrated away from the Jetty logging facade to slf4j 2.0

As for your usage of Configuration.ClassList ...

https://github.com/arquillian/arquillian-container-jetty/blob/12ad61422f25ae4af693bca5705c663c73adfd0f/jetty-embedded-9/src/main/java/org/jboss/arquillian/container/jetty/embedded_9/JettyEmbeddedContainer.java#L144-L166

There's many restrictions to usage of that class in Jetty 9.
First, you must be aware of the order of Configurations, as it's very important.
Next, you should never add duplicate Configuration entries. (since you just use `.add()` you can get duplicates very easily)
This pretty much means that the first code block (using `containerConfig.getConfigurationClasses()`) has no hope of working properly on Jetty 9 for a typical user (even the Jetty devs would find it tricky to near impossible to use that feature on arquillian properly).  I would encourage you to removed that feature entirely in the `embedded_9/JettyEmbeddedContainer.java`.

For Jetty 10+ (yes, there's a Jetty 10.0.0 supporting Servlet 4.0), the `Configuration` objects are self-ordering (and self discovering).
Eg: if `jetty-annotations-11.0.0.jar` is present on the classpath, the [`META-INF/services/org.eclipse.jetty.webapp.Configuration`](https://github.com/eclipse/jetty.project/blob/jetty-11.0.0/jetty-annotations/src/main/resources/META-INF/services/org.eclipse.jetty.webapp.Configuration) is loaded by the `ServiceLoader`, which finds and loads `org.eclipse.jetty.annotations.AnnotationConfiguration` automatically adds it to the server Configuration list.

**Comment by manovotn** [comment]
Thanks for reply. I'm hoping to find some time next week to look into this again.

**Comment by starksm64** [comment]
Closing as there is a jetty-embedded-11 module now
